capstone/src/main/webapp/scripts/post.js

```diff
@@ -0,0 +1,193 @@
+function init() {
+  loadPosts();
+  uploadImage();
+  //fetchBlobstoreUrlAndShowForm();
+}
+
+function uploadImage() {
+  console.log("in upload image");
+  document.getElementById('camera-btn').addEventListener('click', openDialog);
+    function openDialog() {
+      console.log("in open dialogue");
+        document.getElementById('fileid').click();
+        document.getElementById('fileid').addEventListener('change', fetchBlobstoreUrlAndShowForm);
+
+        function fetchBlobstoreUrlAndShowForm() {
+          console.log("in fetch");
+          fetch('/post-image-handler')
+          .then((response) => {
+            return response.text();
+          })
+          .then((imageUploadUrl) => {
+            const messageForm = document.getElementById('post-form');
+            messageForm.action = imageUploadUrl;
+            console.log(imageUploadUrl);
+          });
+        }
+    }
+}
+
+function loadPosts() {
+  fetch('/group-post').then(response => response.json()).then((postsResp) => {
+    const posts = postsResp.posts;
+    const allPostsList = document.getElementById('posts-container');
+    allPostsList.innerHTML = '';
+    for (var i = 0; i < posts.length; i++) {
+      allPostsList.appendChild(createSinglePost(posts[i]));
+    }
+  }).then(() => {
+    var elements = document.getElementsByClassName('post-btn align-vertical comment-btn');
+    for (var i = 0; i < elements.length; i++) {
+      elements[i].addEventListener("click", function() {
+        postComment(this.id, this.id + "comment-input")
+      });
+    }
+
+    var likeBtns = document.getElementsByClassName('like-icon vertical-align');
+    for (var i = 0; i < likeBtns.length; i++) {
+      likeBtns[i].addEventListener("click", function() {
+        //this.classList.toggle("active");
+        likeToggled(this.id, true);
+      });
+    }
+  });
+}
+
+function likeToggled(likeId, liked) {
+  const postId = likeId.substring(0, likeId.length - 4);
```

Comment:
Is it guaranteed that the postId is always going to be the first part of the likeId? I'm just not familiar with how you're getting the values, but if you're constructing the ids yourself this should be fine

Reply:
I'm constructing the likeId to be the postId + "like" so I'm using substring to just get the postId.